packages/features/form-builder/FormBuilder.tsx

```diff
@@ -119,29 +125,37 @@ export const FormBuilder = function FormBuilder({
         <p className="text-subtle mt-0.5 max-w-[280px] break-words text-sm sm:max-w-[500px]">{description}</p>
         <ul ref={parent} className="border-subtle divide-subtle mt-4 divide-y rounded-md border">
           {fields.map((field, index) => {
-            const options = field.options
-              ? field.options
-              : field.getOptionsAt
-              ? dataStore.options[field.getOptionsAt as keyof typeof dataStore]
-              : [];
-
-            // Note: We recently started calling getAndUpdateNormalizedValues in the FormBuilder. It was supposed to be called only on booking pages earlier.
-            // Due to this we have to meet some strict requirements like of labelAsSafeHtml.
+            let options = field.options ?? null;
+            const sources = [...(field.sources || [])];
+            const isRequired = isRequiredField(field);
+            if (!options && field.getOptionsAt) {
+              options = dataStore.options[field.getOptionsAt as keyof typeof dataStore] ?? [];
+              // TODO: The dataStore should itself provide the label directly
+              // This is important because FormBuilder isn't aware of what a location is. It is a generic Form Builder
+              const sourceLabel = field.getOptionsAt === "locations" ? "Location" : field.getOptionsAt;
+              options.forEach((option) => {
+                sources.push({
+                  id: option.value,
+                  label: sourceLabel,
+                  type: "system",
+                });
+              });
```

Comment:
![image](https://github.com/user-attachments/assets/4e36a0dc-960e-4fbb-a3fa-ff1d1d942c0c)
This shows a badge that this field is coming due to locations. This is similar to what we do on fields added through workflows